### DISCUSSION TITLE
WhereIn converting int32 values to question marks???

### DISCUSSION THREAD

**Discussion by the-real-chris-kin** [sqlkata/querybuilder#659] [Q&A] [answered]
The call to WhereIn is pretty simple: 

filterModel.Value is of type object

filterModel.Value is assigned a List<int> before reaching this code

`IEnumerable<int> ids = filterModel.Value as IEnumerable<int>;`
`query.WhereIn(filterModel.FullyQualifiedFieldName, ids);`

But for some reason the resulting condition in my SQL string has (?, ?) instead of (1, 2). Using intellisense I can see that everything in the query object looks correct, the values are integers.

Any insight as to why the values are being output as question marks?

EDIT

Just for funsies I tried the following

`IEnumerable<int> ids = filterModel.Value as IEnumerable<int>;`
`IEnumerable<string> stringIds = ids.Select(x => x.ToString());`
`IEnumerable<int> newIds = stringIds.Select(x => int.Parse(x));`
`query = query.WhereIn(filterModel.FullyQualifiedFieldName, ids);`
`query = query.WhereIn(filterModel.FullyQualifiedFieldName, stringIds);`
`return query.WhereIn(filterModel.FullyQualifiedFieldName, newIds);`

and that produced:

WHERE [A].[StatusID] IN (?, ?) 
  AND [A].[StatusID] IN (?, ?) 
  AND [A].[StatusID] IN (?, ?)

**Comment by the-real-chris-kin** [comment, marked as answer]
I believe I found the cause. The query was paramtizing all the values being used in the where clauses, so looking at the CompiledQuery.SQL you can see those where clauses as WHERE [A].[StatusID] IN (@p0, @p1)

I am building this query entirely free of outside user input so I don't have to fear injection concerns, so I can work around this by replacing the params with their values in the resulting SQL string. If there isn't a way to turn off that parameterization that would be a nice feature to add.

**Reply by ahmad-moussawi** [reply]
You can use `SqlResult.ToString()`. It automatically does what you want, but I don't recommend that at all; even when you don't care about SQL injection, it's nice to keep the params binding. It helps in query caching and speeds up execution. and keep you safe if you use some user input later.
